templates/CRM/Contact/Import/Form/MapTable.tpl

```diff
@@ -29,127 +29,93 @@
-                  {if $wizard.currentStepName == 'Preview'}
-                    {if $relatedContactDetails && $relatedContactDetails[$i] != ''}
-                            {$mapper[$i]} - {$relatedContactDetails[$i]}
-
-                            {if $relatedContactLocType && $relatedContactLocType[$i] != ''}
-                              - {$relatedContactLocType[$i]}
-                      {/if}
-
-                            {if $relatedContactPhoneType && $relatedContactPhoneType[$i] != ''}
-                              - {$relatedContactPhoneType[$i]}
-                      {/if}
-
-                            {* append IM Service Provider type for related contact *}
-                            {if  $relatedContactImProvider && $relatedContactImProvider[$i] != ''}
-                                - {$relatedContactImProvider[$i]}
-                            {/if}
-
-          {* append website type *}
-          {if  $relatedContactWebsiteType && $relatedContactWebsiteType[$i] != ''}
-                                - {$relatedContactWebsiteType[$i]}
-                            {/if}
-
-          {else}
-
-                            {if $locations[$i]}
-                                {$locations[$i]} -
-                            {/if}
-
-                            {if $phones[$i]}
-                                {$phones[$i]} -
-                            {/if}
-
-                            {* append IM Service provider type for contact *}
-                            {if $ims[$i]}
-                                {$ims[$i]} -
-                            {/if}
-
-          {* append website type *}
-                            {if $websites[$i]}
-                                {$websites[$i]} -
-                            {/if}
-
-                            {*else*}
-                                {$mapper[$i]}
-                            {*/if*}
-                        {/if}
-                  {else}
-                    {$mapperField.html|smarty:nodefaults}
-                  {/if}
-                </td>
-
-            </tr>
+    {*Loop on columns parsed from the import data rows*}
+    {foreach from=$mapper key=i item=mapperField}
+      <tr style="border: 1px solid #DDDDDD;">
+        {if array_key_exists($i, $columnNames)}
+          <td class="even-row labels">{$columnNames[$i]}</td>
+        {/if}
+        {foreach from=$dataValues item=row key=index}
+          <td class="odd-row">{$row[$i]|escape}</td>
         {/foreach}
 
-    </table>
+        {* Display mapper <select> field for 'Map Fields', and mapper value for 'Preview' *}
+        <td class="form-item even-row{if $wizard.currentStepName == 'Preview'} labels{/if}">
+          {if $wizard.currentStepName == 'Preview'}
+            {if $relatedContactDetails && $relatedContactDetails[$i] != ''}{$mapper[$i]} - {$relatedContactDetails[$i]}
+              {if $relatedContactLocType && $relatedContactLocType[$i] != ''} - {$relatedContactLocType[$i]}{/if}
```

Comment:
I find this harder to read than the original. Is this being done because the linebreaks affect the output?

e.g. this seems clearer to me:

```
            {if $relatedContactDetails && $relatedContactDetails[$i] != ''}
              {$mapper[$i]} - {$relatedContactDetails[$i]}
```

Reply:
@demeritcowboy hmm - I went with it because you can see all the ifs lined up & each one doesn't put much in - however, maybe that if is more important than the others so deserves the extra space - since the mapper is a bit 'more'

FYI - I'm hoping to move that work to the php layer

Reply:
Line 46 is also hard to read for the same reason - it looks like the minus sign is a subtraction operation.

Reply:
@demeritcowboy I fixed up the first one - but am less inclined to for the rest - because they all follow that same pattern of adding a - + one word & being able to skim the if's seems really useful

Reply:
Ok. It's hard to get consensus on visual preferences. Maybe we could do like wikipedia and make some bots that just keep editing each other's edits, so that all the readers are happy at least some of the time (grin).

Reply:
@demeritcowboy hopefully this code is on it's way out anyway - it was just doing my head in & once I re-laid it out I could see where the ifs started & finished